chore(docs): simplify "Create tags pages" documentation (#15576)
gatsbyjs/gatsby: docs/docs/adding-tags-and-categories-to-blog-posts.md

```graphql
{
  allMarkdownRemark {
    group(field: frontmatter___tags) {
      tag: fieldValue
      totalCount
    }
  }
}
```

> The above query groups posts by `tags`, and return each `tag` with the number of posts for each as `totalCount`. We could also extract some post data in each group, but to create a page for each tag, we only need to know the tag name. Let's make the tag page template now:

Reviewer: The sentence has a lot of small words and makes it hard to read in my opinion.

I'm suggesting:
```suggestion
The above query groups posts by `tags`, and returns each `tag` with the number of posts as `totalCount`. As an addition, we could extract some post data in each group if we need to. To keep this tutorial small, we're only using the tag name in our tag pages. Let's make the tag page template now:
```

Now we've got a template. Great! I'll assume you followed the tutorial for [Adding Markdown Pages](/docs/adding-markdown-pages/) and provide a sample `createPages` that generates post pages as well as tag pages. In the site's `gatsby-node.js` file, include `lodash` (`const _ = require('lodash')`) and then make sure your [`createPages`](/docs/node-apis/#createPages) looks something like this:

> ```js

Reviewer: Any reason to delete the title of the file? People benefit from this small metadata as it helps them navigate through the source and making sure they are in the right file.

![image](https://user-images.githubusercontent.com/1120926/61691299-864e7b00-ad2b-11e9-9c8b-9cfbd18a776c.png)

Author: No good reason, no. I removed it to get JS syntax highlighting working in my editor and forgot to add it again. Should be fixed now.

  return graphql(`
    {

>       poatsRemark: allMarkdownRemark(

Reviewer 2: did you mean `posts`?
```suggestion
      postsRemark: allMarkdownRemark(
```

Author: Of course 😁

          }
        }
      }
      tagsGroup: allMarkdownRemark(limit: 2000) {
        group(field: frontmatter___tags) {
          fieldValue
        }
      }
    }
  `).then(result => {
    if (result.errors) {
      return Promise.reject(result.errors)
    }

>     const posts = result.data.poatsRemark.edges

Reviewer 2: ```suggestion
    const posts = result.data.postsRemark.edges
```

Some notes:

- Our GraphQL query only looks for data we need to generate these pages. Anything else can be queried again later (and, if you notice, we do this above in the tags template for the post title).

> - We have 2 `allMarkdownRemark` bits so we must alias one of them. We alias both to avoid generic names their content.

Reviewer: ```suggestion
- We have referenced two `allMarkdownRemark` fields in our query. To avoid naming collisions we must alias one of them. We alias both to make our code more human-readable.
```

Some notes:

- Our GraphQL query only looks for data we need to generate these pages. Anything else can be queried again later (and, if you notice, we do this above in the tags template for the post title).

> - We have referenced two `allMarkdownRemark` fields in our query. To avoid naming collisions we must alias one of them. We alias both to make our code more human-readable.

Reviewer 3: ```suggestion
- We have referenced two `allMarkdownRemark` fields in our query. To avoid naming collisions we must [alias](/docs/graphql-reference/#aliasing) one of them. We alias both to make our code more human-readable.
```

Linking to this could help some people who haven't heard of GraphQL aliases 🙂
